layers/+distribution/spacemacs-base/packages.el

```diff
@@ -847,7 +852,23 @@ below. Anything else exits."
       (add-to-list 'recentf-exclude
                    (expand-file-name spacemacs-cache-directory))
       (add-to-list 'recentf-exclude (expand-file-name package-user-dir))
-      (add-to-list 'recentf-exclude "COMMIT_EDITMSG\\'"))))
+      (add-to-list 'recentf-exclude "COMMIT_EDITMSG\\'")
+
+      (defsubst spacemacs//file-was-viewed-p (file)
+        "Return non-nil if FILE's buffer exists and has been displayed."
+        (interactive)
+        (let ((buf (find-buffer-visiting file)))
+          (> (buffer-local-value 'buffer-display-count buf) 0)))
+
+      (defsubst spacemacs//recentf-viewed-predicate (file)
+        "Return non-nil if FILE was previously in recentf list or has been
+displayed during this session."
+        (and (recentf-keep-default-predicate file)
+             (or (member file spacemacs-recentf-on-load)
+                 (spacemacs//file-was-viewed-p file))))
+
+      ;; And, of course, you now need:
+      (setf recentf-keep '(spacemacs//recentf-viewed-predicate)))))
```

Comment:
why `setf` and not `add-to-list`? By overriding the existing value, we risk interfering with user's personal configuration.